src/sdltiles.cpp

```diff
@@ -851,14 +851,67 @@ void cata_cursesport::curses_drawwindow( const catacurses::window &w )
     WINDOW *const win = w.get<WINDOW>();
     bool update = false;
     if (g && w == g->w_terrain && use_tiles) {
+        // Strings with colors do be drawn with map_font on top of tiles.
+        std::multimap<point, formatted_text> overlay_strings;
+
         // game::w_terrain can be drawn by the tilecontext.
         // skip the normal drawing code for it.
         tilecontext->draw(
             win->x * fontwidth,
             win->y * fontheight,
             tripoint( g->ter_view_x, g->ter_view_y, g->ter_view_z ),
             TERRAIN_WINDOW_TERM_WIDTH * font->fontwidth,
-            TERRAIN_WINDOW_TERM_HEIGHT * font->fontheight);
+            TERRAIN_WINDOW_TERM_HEIGHT * font->fontheight,
+            overlay_strings );
+
+        point prev_coord;
+        int x_offset = 0;
+        for( const auto &iter : overlay_strings ) {
+            const point coord = iter.first;
+            const formatted_text ft = iter.second;
+            const utf8_wrapper text( ft.text );
+            const auto wtext = utf8_to_wstr(ft.text);
+
+            // Strings at equal coords are displayed sequentially.
+            if( coord != prev_coord ) {
+                x_offset = 0;
+            }
+
+            // Calculate length of all strings in sequence to align them.
+            int full_text_length = 0;
+            const auto range = overlay_strings.equal_range( coord );
+            for( auto ri = range.first; ri != range.second; ++ri ) {
+                utf8_wrapper rt( ri->second.text );
+                full_text_length += rt.display_width();
+            }    
+
+            int alignment_offset = 0;
+            if ( ft.alignment == TEXT_ALIGNMENT_CENTER ) {
+                alignment_offset = full_text_length / 2;
+            } else if ( ft.alignment == TEXT_ALIGNMENT_RIGHT ) {
+                alignment_offset = full_text_length - 1;
+            }
+
+            // TODO: ensure it works with composite chars
```

Comment:
Is this still in question?

Reply:
Checked how it works, my code passes correct substrings to `OutputChar`, but it still renders incorrectly. It does render the same way in other parts of the game though.